solver/src/main/java/org/chocosolver/solver/search/strategy/Search.java

```diff
@@ -153,6 +150,33 @@ public static SetStrategy setVarSearch(SetVar... sets) {
         return setVarSearch(new GeneralizedMinDomVarSelector(), new SetDomainMin(), true, sets);
     }
 
+    // ************************************************************************************
+    // GRAPHVAR STRATEGIES
+    // ************************************************************************************
+
+    /**
+     * Lexicographic graph branching strategy.
+     * Branch on nodes then edges.
+     * <br>
+     * <br> node branching:
+     * Let i be the first node such that
+     * i in envelope(g) and i not in kernel(g).
+     * The decision adds i to the kernel of g.
+     * It is fails, then i is removed from the envelope of g.
+     * <br>
+     * edge branching:
+     * <br> node branching:
+     * Let (i,j) be the first edge such that
+     * (i,j) in envelope(g) and (i,j) not in kernel(g).
+     * The decision adds (i,j) to the kernel of g.
+     * It is fails, then (i,j) is removed from the envelope of g
+     *
+     * @param g a graph variable to branch on
+     */
+    public static GraphStrategy graphVarLexSearch(GraphVar g) {
```

Comment:
We could consider, in another PR, to change this name to a shorter one

Reply:
Yes, would you like to do it now, or should we wait until the moment in which we will focus on graph strategies?

Reply:
I can wait until you work on strategies